lrlex/src/lib/ctbuilder.rs

```diff
@@ -815,9 +817,15 @@ where
         };
         // Try and run a code formatter on the generated code.
         let unformatted = out_tokens.to_string();
-        let outs = syn::parse_str(&unformatted)
+        let mut outs = String::new();
+        // Record the time that this version of lrlex was built. If the source code changes and rustc
+        // forces a recompile, this will change this value, causing anything which depends on this
+        // build of lrlex to be recompiled too.
+        let timestamp = env!("VERGEN_BUILD_TIMESTAMP");
```

Comment:
Is this environment variable always defined?

Reply:
Yes, you can see it also used in the `ct_token_map` codegen at the top of the diff context in the first patch in this series.  [719da09](https://github.com/softdevteam/grmtools/pull/576/commits/719da0915c789b4d5a9de1f3c627a5903eee07a7)

I believe it is ensured by:
https://docs.rs/vergen/latest/vergen/struct.BuildBuilder.html#method.build_timestamp
https://github.com/softdevteam/grmtools/blob/master/lrlex/build.rs#L4

along with the weirdness of the `env!` macro:
https://www.cs.brandeis.edu/~cs146a/rust/doc-02-21-2015/std/macro.env!.html

So I think this is including a timestamp from the time when `lrlex/build.rs` is built to every `token_map.rs`,
so this addition syncs things so that applies to the main lex output as well!

Reply:
Right, yes, I can see. @ratmice I think you're best placed to judge this PR.